src/tmpfiles/tmpfiles.c

```diff
@@ -1276,9 +1276,18 @@ static int create_item(Item *i) {
                                 log_debug("Quota for subvolume \"%s\" already in place, no change made.", i->path);
                 }
 
-                /* fall through */
+                r = path_set_perms(i, i->path);
+                if (q < 0)
+                        return q;
+                if (r < 0)
+                        return r;
+
+                break;
 
         case EMPTY_DIRECTORY:
+                if (access(i->path, F_OK) < 0)
+                        break;
+
```

Comment:
Hmm, explicit access() checks like this always have this "racy" feel, as things might change between the access() and the application of the perms. In this case this isn't a real issue admittedly, but it still feels like something to avoid. Moreover, I'd prefer if we'd only skip the permission stuff only if the file doesn't exsit, but not under other error cases (for example access() might fail with EACCES if some leading component of the path has too strict access modes)...

Hence, I think it would be much nicer if we'd call path_set_perms() unconditionally here, but then check for it  ENOENT and ignoring the whole thing for the "e" case...

(admittedly, the patch is not going to be as trivial as suggested here, as path_set_perms() needs to use different log levels for its messages dependening on whetger the operation is "e" or some other operation, but this shouldn't be too difficult to get right, simply use log_full() and check `i->type` and errno when determining which log level to use, i.e. LOG_ERROR or LOG_DEBUG, if you follow what I mean)

Reply:
Initially I thought about doing it this way, but ```access()``` check was simpler and ```path_set_perms() ```is called in my other contexts so I didn't want to complicate the function just of because "e" case.

Reply:
Still, I don't like the racy feel, and I don't like ignoring any other errors than ENOENT here...